pkg/integrations/v4/protocol/types.go

```diff
@@ -129,6 +130,81 @@ type MetricData map[string]interface{}
 // EventData is the data type for single shot events
 type EventData map[string]interface{}
 
+// NewEventData create a new event data from builder func
+func NewEventData(options ...func(EventData)) (EventData, error) {
+	e := EventData{
+		"eventType": "InfrastructureEvent",
+		"category":  "notifications",
+	}
+
+	for _, opt := range options {
+		opt(e)
+	}
+
+	// Validate required field
+	if _, ok := e["summary"]; !ok {
+		return nil, errors.New("invalid event format: missing required 'summary' field")
+	}
+
+	// there are integrations that add the hostname
+	// and since backed has a attribute limit
+	// we remove it to avoid potential conflict when submitting events
+	delete(e, "hostname")
+
+	return e, nil
+}
+
+// Builder for NewEventData constructor will copy only valid keys
+// valid keys: ["summary", "category", "entity_name", "format", "local_identity", "local_details"]
+func WithEvents(original EventData) func(EventData) {
+	return func(copy EventData) {
+		acceptedAttribute := []string{"summary", "category", "entity_name", "format", "local_identity", "local_details"}
```

Comment:
Might be worth moving into a `var` at the top of the file. Apart from that all looks good.

Reply:
Done, thanks